var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -102,8 +102,8 @@ class Phist(CMakePackage):
 
     # ###################### Patches ##########################
 
+    patch('ppc64_sse.patch', when='@1.7.4:1.9.4')
```

Comment:
Can you add a brief comment to explain why this is needed and maybe point to its provenance? A one liner is fine, it's just to have some clue in the package.py file.

Reply:
While at it, as PPC isn't the only architecture family without X86 SSE intrinsics:
 
Maybe it becomes more obvious (and also semantically correct), if the patch isn't like this:
```py
+if(PHIST_HOST_OPTIMIZE)
+  if (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "ppc64le")
+    add_definitions(-DNO_WARN_X86_INTRINSICS)
+  endif()
+endif()
```
but rather like:
```py
+if(PHIST_HOST_OPTIMIZE)
+  if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+    add_definitions(-DNO_WARN_X86_INTRINSICS)
+  endif()
+endif()
```
This also makes sense as `aarch64` is also used with spack and maybe `RISCV` in the future.

Caution: Didn't test this check, it's based on internet search:
https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html
https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html
https://github.com/opencv/opencv/blob/1261f250c6cb33e411d257406e6ac718bed6fdc3/cmake/OpenCVDetectCXXCompiler.cmake#L87
With the `NOT` added in front (yes it's weird, but so is the CMAKE if NOT inversion AFAIK.